addons/website/models/res_users.py

```diff
@@ -87,3 +87,12 @@ def authenticate(cls, db, login, password, user_agent_env):
                 visitor_pre_authenticate_sudo.access_token = user_partner.id
                 visitor_pre_authenticate_sudo._update_visitor_last_visit()
         return uid
+
+    def write(self, vals):
+        # If the user was a portal and becomes internal, remove website_id.
+        portal_users = self.filtered(lambda user: user._is_portal())
```

Comment:
You can add a `website_id` check here (or in the next lambda) to filter out the ones not related to a website, it's useless to write `website_id=False` on a recordset where most of the time those won't have a website_id anyway.
The check will lead the write to a noop as no record.
```suggestion
        portal_users = self.filtered(lambda user: user._is_portal() and user.website_id)
```